src/tree_schema_helpers.c

```diff
@@ -1141,3 +1141,198 @@ lysp_find_module(struct ly_ctx *ctx, const struct lysp_module *mod)
     return NULL;
 }
 
+enum yang_keyword
+match_keyword(const char *data)
```

Comment:
Don't you miss the length of the input data? What if `lyxml_get_element()` reads `name` as `typex` with length 5? You accespt it as `type` keyword.